compiler/rustc_hir/src/hir.rs

```diff
@@ -3356,13 +3353,14 @@ impl<'hir> OwnerNode<'hir> {
         }
     }
 
-    pub fn span(&self) -> Span {
+    #[allow(rustc::pass_by_value)]
+    pub fn span(&self) -> &'hir Span {
```

Comment:
Should we both take and return by value?

Reply:
It's returned by reference because it's used by reference in `Node::Err`.

Reply:
It should still return `Span`, we should minimize the places where `&Span` is used.

Reply:
`Node::Err` needs `&'hir Span` with `'hir` lifetime specifically, and this function provides it.
It cannot return `Span` by value, unless we use something else (like manual reimplementation of this same function), for giving a span to `Node::Err`.

Reply:
Ah, you then need to create a reference. Makes sense.